src/Ingester/Ingester/UserIngester.php

```diff
@@ -15,77 +15,51 @@
  *  along with this program; if not, write to the Free Software
  *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- *  Copyright (c) 2019 (original work) Open Assessment Technologies S.A.
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies S.A.
  */
 
 declare(strict_types=1);
 
 namespace OAT\SimpleRoster\Ingester\Ingester;
 
-use Doctrine\Persistence\ManagerRegistry;
-use Exception;
-use OAT\SimpleRoster\Entity\Assignment;
-use OAT\SimpleRoster\Entity\EntityInterface;
-use OAT\SimpleRoster\Entity\User;
-use OAT\SimpleRoster\Exception\LineItemNotFoundException;
-use OAT\SimpleRoster\Model\LineItemCollection;
-use OAT\SimpleRoster\Repository\LineItemRepository;
+use OAT\SimpleRoster\DataTransferObject\AssignmentDtoCollection;
+use OAT\SimpleRoster\DataTransferObject\UserDtoCollection;
+use OAT\SimpleRoster\Repository\NativeAssignmentRepository;
+use OAT\SimpleRoster\Repository\NativeUserRepository;
 
-class UserIngester extends AbstractIngester
+class UserIngester
 {
-    /** @var LineItemCollection */
-    private $lineItemCollection;
+    /** @var NativeUserRepository */
+    private $userRepository;
 
-    /** @var LineItemRepository */
-    private $lineItemRepository;
+    /** @var NativeAssignmentRepository */
+    private $assignmentRepository;
 
-    public function __construct(LineItemRepository $lineItemRepository, ManagerRegistry $managerRegistry)
+    public function __construct(NativeUserRepository $userRepository, NativeAssignmentRepository $assignmentRepository)
     {
-        $this->lineItemRepository = $lineItemRepository;
-
-        parent::__construct($managerRegistry);
+        $this->userRepository = $userRepository;
+        $this->assignmentRepository = $assignmentRepository;
     }
 
-    public function getRegistryItemName(): string
+    public function ingest(UserDtoCollection $users): void
```

Comment:
Suggestion: we could create another class (AssignmentIngester) and move the related logic to this new class and on the UserIngester Command could receive it as a dependency to make the assignments ingestion based on the users. At least for this context, this class wouldn't have more than one responsibility.

Reply:
I think it will be handled on RO-53

Reply:
Yes, this will be done under that ticket (already in progress)